demo/screens/ListLandsScreen.js

```diff
@@ -19,39 +21,45 @@ export default class ListLandsScreen extends React.Component {
     this.setState({ rows });
   }
 
+  // Note: This function is assuing that:
+  // - All estates have a sell order
+  // - The total supply of estates is small
+  // TODO: Rewrite this function when we move to testnet
   async getSellOrders() {
     const orders = [];
-    const estateContract = new Contract(estateAddress, estateABI);
-    const totalSupply = await estateContract.totalSupply();
+    const totalSupply = await this.estateContract.totalSupply();
 
-    for (let estateId = 1; estateId <= Number(totalSupply); estateId++)
-      orders.push(this.getSellOrder(estateId));
+    for (let estateId = 1; estateId <= Number(totalSupply); estateId++) {
+      const order = this.getSellOrder(estateId);
+      orders.push(order);
+    }
 
     return await Promise.all(orders);
   }
 
   async getSellOrder(estateId) {
-    const estateContract = new Contract(estateAddress, estateABI);
-    const marketplaceContract = new Contract(marketplaceAddress, markplaceABI);
-
-    const estateName = await estateContract.getMetadata(estateId);
+    const estateName = await this.estateContract.getMetadata(estateId);
     const [
       orderId,
       seller,
       price,
       expiresAt,
-    ] = await marketplaceContract.auctionByAssetId(estateId);
+    ] = await this.marketplaceContract.auctionByAssetId(estateId);
+
+    const hasOrder = parseInt(orderId, 16) !== 0;
+    if (!hasOrder) throw Error(`Estate (id:${estateId}) has no sell order.`);
 
     const priceMana = Number(price.toString()) / 1e18;
-    const manaUsdRate = 30;
-    const priceUsd = priceMana * manaUsdRate;
+    const manaPerUsd = 30;
+    const priceUsd = priceMana * manaPerUsd;
```

Comment:
Hmm wait, wouldn't the price in USD be a lower number than the price in mana? So should we divide by 30?

Reply:
Oh yes, my fault! Done.

Reply:
Thanks! Resolving!